google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/DocumentReference.java

```diff
@@ -226,7 +185,7 @@ public T apply(List<T> results) {
    * @return An ApiFuture that will be resolved when the write finishes.
    */
   @Nonnull
-  public ApiFuture<WriteResult> update(@Nonnull Map<String, Object> fields) {
+  public ApiFuture<WriteResult> update(@Nonnull Object fields) {
```

Comment:
Updates still needs to take in a Map of <String,Object> and should not accept top-level POJOs (nested POJOs are ok). The reason for this is that `update()` treats the keys of the `fields` map as field paths and not as string literals. If you passed a POJO with a property name of "foo.bar", `update()` would actually updated the nested field "bar" in map "foo".